# Import Libraries

In [ ]:
# Install selenium bs4
!pip install selenium bs4

In [3]:
# Import library webdriver, Beautifulsoup, Pandas
from selenium import webdriver
from bs4 import BeautifulSoup
import pandas as pd

# Extract

- Web Scraping

Tentukan data yang ingin diambil dan jalankan driver (bot) untuk membuka dan mengambil data dari web yang dituju, simpan di dalam soup

In [30]:
driver = webdriver.Chrome() # Jalankan driver
# Membuat list kosong tiap kolom dan mendefinisikan nama tiap kolom
product        = []
price          = []
sold           = []
rating         = []
image          = []
link           = []       

for page in range(1,12):    # Menentukan range halaman yang ditentukan
    driver.get(f'https://ottencoffee.co.id/categories/coffee?page={page}') # Menjadikan page semula menjadi range page yang diinginkan
    driver.maximize_window() # Membuat jendela popup maximize
    scheight = .1  
    while scheight < 9.9:    # Membiarkan driver secara otomatis melakukan scroll agar data terbaca satu halaman full dengan while loop
            driver.execute_script("window.scrollTo(0, document.body.scrollHeight/%s);" % scheight)
            scheight += .0015    

    html = driver.page_source   # Menentukan bahwa html adalah page_source dari driver
    soup = BeautifulSoup(html, 'html.parser')   # Menjadikan soup sebagai data yang berhasil diambil
    # Scraping data dari tokopedia seblak
    for data in soup.find_all('div', {'class':'sc-drKuOJ jueqIz'}): # Menentukan kotak per produk
        try:
            product.append(data.find('div', {'class':'sc-fnwBNb gqOztv title-product'}).get_text())
        except:
            product.append(None)
        try:
            price.append(data.find('div', {'class':'sc-dnqmqq iSQPAv'}).get_text())
        except:
            price.append(None)
        try:
            sold.append(data.find('span', {'class':'sc-bZQynM licOZQ eo-color-gray gsb-flex center-text'}).get_text())
        except:
            sold.append(None)
        try:
            rating.append(data.find('div', {'class':'gsb-flex'}).get_text())
        except:
            rating.append(None)
        try:
            image.append(data.find('img', {'class':'sc-cugefK bhRvD product-img'})['src'])
        except:
            image.append(None)
        try:
            link.append(data.find('a')['href'])
        except:
            link.append(None)
driver.close()  # Tutup driver

# Data Checking

In [26]:
# Melihat apakah sudah jumlah data yang ada sudah sesuai atau belum
print(len(product))
print(len(price))
print(len(sold))
print(len(rating))
print(len(image))
print(len(link))

195
195
195
195
195
195


# Data Transformation

 Mendefinisikan DataFrame yang digunakan dan mendeklarasikannya per kolom

In [31]:
df = pd.DataFrame()

df['Product']       = product
df['Price']         = price
df['Rating']        = rating
df['Sold']          = sold
df['Image']         = image
df['Link']          = link

In [32]:
# Memastikan apakah DataFrame berhasil didefinisikan dan dideklarasikan
df

Product       Price  \
0              Crema Espresso - Kopi House Blend 500gr  Rp 111.600   
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  Rp 151.200   
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  Rp 143.100   
3                              Kopi Susu Blend - 500gr   Rp 83.700   
4                              Kopi Gold Blend - 500gr   Rp 93.600   
..                                                 ...         ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   Rp 78.000   
191      Refill Cold Brew - Arabica Mandheling Natural   Rp 78.000   
192        Refill Cold Brew - Robusta Bali Full Washed   Rp 78.000   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  Rp 198.000   
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  Rp 198.000   

          Rating             Sold  \
0    4.9 (2280)  326,1rb Terjual   
1     4.9 (119)   18,2rb Terjual   
2     4.9 (137)   15,5rb Terjual   
3     4.9 (902)  137,2rb Terjual   
4     4.9 (637)   90,1rb Terjual   
..           ...              ...   
190         None             None   
191       5 (3)      172 Terjual   
192         None             None   
193       5 (2)      107 Terjual   
194       5 (2)      105 Terjual   

                                                 Image  \
0    https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1    https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2    https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4    https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..                                                 ...   
190  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Product  195 non-null    object
 1   Price    195 non-null    object
 2   Rating   174 non-null    object
 3   Sold     173 non-null    object
 4   Image    195 non-null    object
 5   Link     195 non-null    object
dtypes: object(6)
memory usage: 9.3+ KB


Berdasarkan data di atas, DataFrame df memiliki 195 rows yang terdiri dari row 0-194. Serta memiliki 6 kolom yang terdiri dari:
1. Kolom Product, memiliki 195 data non-null yang berarti tidak terdapat missing value
2. Kolom Price, memiliki 195 data non-null yang berarti tidak terdapat missing value
3. Kolom Rating, memiliki 174 data non-null yang berarti memiliki 174 data tidak missing value dan terdapat 21 data yang missing valuenya
4. Kolom Sold, memiliki 173 data non-null yang berarti memiliki 173 data tidak missing value dan terdapat 22 data yang missing valuenya
5. Kolom Image, memiliki 195 data non-null yang berarti tidak terdapat missing value
6. Kolom Link, memiliki 195 data non-null yang berarti tidak terdapat missing value
    
 - Semua tipe data yang ada pada DataFrame ini masih object, yang berarti String

Berdasarkan dua data di atas, DataFrame telah berhasil dideklarasikan dan didefinisikan, namun terdapat beberapa hal pada kolom Price, Rating, dan Sold yang masih belum sesuai

In [34]:
# Menghilangkan objek di depan angka pada kolom Rating
df.Rating = df.Rating.str.replace('','', regex=False)

In [35]:
df['Rating']

0      4.9 (2280)
1       4.9 (119)
2       4.9 (137)
3       4.9 (902)
4       4.9 (637)
          ...    
190          None
191         5 (3)
192          None
193         5 (2)
194         5 (2)
Name: Rating, Length: 195, dtype: object

Berdasarkan data di atas, objek di depan angka berhasil dihilangkan, akan tetapi masih terdapat tanda kurung dan angka dibelakangnya

In [36]:
# Menghilangkan tanda kurung beserta angka di dalamnya
df.Rating = df.Rating.str.replace(r"\(\d+\)?", "", regex=True)

In [37]:
df['Rating']

0      4.9 
1      4.9 
2      4.9 
3      4.9 
4      4.9 
       ... 
190    None
191      5 
192    None
193      5 
194      5 
Name: Rating, Length: 195, dtype: object

Berdasarkan data di atas, data berhasil dibersihkan, sehingga tersisa angka yang diinginkan

In [38]:
df

Product       Price Rating  \
0              Crema Espresso - Kopi House Blend 500gr  Rp 111.600   4.9    
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  Rp 151.200   4.9    
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  Rp 143.100   4.9    
3                              Kopi Susu Blend - 500gr   Rp 83.700   4.9    
4                              Kopi Gold Blend - 500gr   Rp 93.600   4.9    
..                                                 ...         ...    ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   Rp 78.000   None   
191      Refill Cold Brew - Arabica Mandheling Natural   Rp 78.000     5    
192        Refill Cold Brew - Robusta Bali Full Washed   Rp 78.000   None   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  Rp 198.000     5    
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  Rp 198.000     5    

                Sold                                              Image  \
0    326,1rb Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1     18,2rb Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2     15,5rb Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    137,2rb Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4     90,1rb Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..               ...                                                ...   
190             None  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191      172 Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192             None  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193      107 Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194      105 Terjual  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

Berdasarkan data di atas, pada kolom sold, terdapat data yang tertulis masih mengandung (rb), sehingga untuk menganalisa data belum dapat dilakukan dengan baik, sehingga tulisan rb akan diganti dengan angka 000, dan juga 00

In [57]:
# Fungsi untuk mengonversi format 'rb' atau angka biasa ke angka
def convert_to_number(text):
    if not text:  # Jika nilai None atau kosong, kembalikan None
        return None
    # Ambil angka sebelum "Terjual"
    text = text.split()[0]
    
    if "rb" in text:  # Jika mengandung 'rb'
        text = text.replace("rb", "").strip()  # Hapus 'rb'
        text = text.replace(",", ".")  # Ganti koma dengan titik
        if "." in text:
            return float(text) * 1000  # Jika desimal
        else:
            return int(text) * 1000  # Jika bilangan bulat
    else:  # Jika tidak mengandung 'rb', langsung konversi ke integer
        return int(text)

# Terapkan fungsi ke kolom 'Penjualan'
df["Sold"] = df["Sold"].apply(
    lambda x: convert_to_number(x) if isinstance(x, str) else None
)

In [58]:
df.Sold

0      326100.0
1       18200.0
2       15500.0
3      137200.0
4       90100.0
         ...   
190         NaN
191       172.0
192         NaN
193       107.0
194       105.0
Name: Sold, Length: 195, dtype: float64

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Product  195 non-null    object 
 1   Price    195 non-null    object 
 2   Rating   174 non-null    object 
 3   Sold     173 non-null    float64
 4   Image    195 non-null    object 
 5   Link     195 non-null    object 
dtypes: float64(1), object(5)
memory usage: 9.3+ KB


In [61]:
df.Price

0      Rp 111.600
1      Rp 151.200
2      Rp 143.100
3       Rp 83.700
4       Rp 93.600
          ...    
190     Rp 78.000
191     Rp 78.000
192     Rp 78.000
193    Rp 198.000
194    Rp 198.000
Name: Price, Length: 195, dtype: object

Berdasarkan data di atas, data berhasil diubah seperti yang diinginkan, akan tetapi masih ada beberapa hal yang masih belum sesuai

Menurut info dari data di atas, kolom Price, Rating, dan juga Sold masih bertipe data object, yang dimana seperti yang diketahui, untuk harga, rating, dan juga jumlah terjual kita butuhkan untuk analisis data, maka akan diubah ke tipe data numeric (integer, float)

In [ ]:
# Menyesuaikan tipe data dan element yang ingin diambil dari kolom Price
df['Price'] = df['Price'].str.replace('\D', '', regex=True).astype(int)

In [63]:
df['Price']

0      111600
1      151200
2      143100
3       83700
4       93600
        ...  
190     78000
191     78000
192     78000
193    198000
194    198000
Name: Price, Length: 195, dtype: int64

In [64]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Product  195 non-null    object 
 1   Price    195 non-null    int64  
 2   Rating   174 non-null    object 
 3   Sold     173 non-null    float64
 4   Image    195 non-null    object 
 5   Link     195 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 9.3+ KB


Berdasarkan dua data di atas, telah berhasil dilakukan pengubahan dan penyesuaian elemen yang ingin diambil dari kolom price, yakni elemen yang diambil hanya angkanya saja, beserta tipe data yang telah diubah menjadi integer

In [65]:
# Menyesuaikan tipe data pada kolom Rating
df['Rating'] = df['Rating'].astype(float)

In [66]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Product  195 non-null    object 
 1   Price    195 non-null    int64  
 2   Rating   174 non-null    float64
 3   Sold     173 non-null    float64
 4   Image    195 non-null    object 
 5   Link     195 non-null    object 
dtypes: float64(2), int64(1), object(3)
memory usage: 9.3+ KB


Berdasarkan data di atas, telah berhasil dilakukan pengubahan tipe data pada kolom Rating yang mulanya object(string) menjadi float

In [71]:
df

Product   Price  Rating  \
0              Crema Espresso - Kopi House Blend 500gr  111600     4.9   
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  151200     4.9   
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  143100     4.9   
3                              Kopi Susu Blend - 500gr   83700     4.9   
4                              Kopi Gold Blend - 500gr   93600     4.9   
..                                                 ...     ...     ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN   
191      Refill Cold Brew - Arabica Mandheling Natural   78000     5.0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  198000     5.0   
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  198000     5.0   

       Sold                                              Image  \
0    326100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1     18200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2     15500  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    137200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4     90100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..      ...                                                ...   
190       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191     172  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193     107  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194     105  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

In [67]:
# Mengganti value None menjadi angka 0 dan membuatnya bertipe data integer
df["Sold"]= df["Sold"].fillna(0).astype(int)

In [68]:
df

Product   Price  Rating  \
0              Crema Espresso - Kopi House Blend 500gr  111600     4.9   
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  151200     4.9   
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  143100     4.9   
3                              Kopi Susu Blend - 500gr   83700     4.9   
4                              Kopi Gold Blend - 500gr   93600     4.9   
..                                                 ...     ...     ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN   
191      Refill Cold Brew - Arabica Mandheling Natural   78000     5.0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  198000     5.0   
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  198000     5.0   

       Sold                                              Image  \
0    326100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1     18200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2     15500  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    137200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4     90100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..      ...                                                ...   
190       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191     172  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193     107  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194     105  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Product  195 non-null    object 
 1   Price    195 non-null    int64  
 2   Rating   174 non-null    float64
 3   Sold     195 non-null    int64  
 4   Image    195 non-null    object 
 5   Link     195 non-null    object 
dtypes: float64(1), int64(2), object(3)
memory usage: 9.3+ KB


Berdasarkan 2 data di atas, dapat disimpulkan bahwa penggantian telah berhail dilakukan.
Tipe data pada kolom Sold yang mulanya object telah berhasil menjadi int

In [70]:
df

Product   Price  Rating  \
0              Crema Espresso - Kopi House Blend 500gr  111600     4.9   
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  151200     4.9   
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  143100     4.9   
3                              Kopi Susu Blend - 500gr   83700     4.9   
4                              Kopi Gold Blend - 500gr   93600     4.9   
..                                                 ...     ...     ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN   
191      Refill Cold Brew - Arabica Mandheling Natural   78000     5.0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  198000     5.0   
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  198000     5.0   

       Sold                                              Image  \
0    326100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1     18200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2     15500  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    137200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4     90100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..      ...                                                ...   
190       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191     172  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193     107  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194     105  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

# Simple Analysis

In [72]:
df[df['Sold'] == 0]

Product   Price  Rating  Sold  \
54        Argopuro Anaerobic Natural 200g Kopi Arabica  106200     5.0     0   
110                         Peaberry 500g Kopi Arabica  169200     4.8     0   
114       Java Ciwidey Honey Process 500g Kopi Arabica  230400     4.9     0   
122  Refill Cold Brew - Arabica Flores Bajawa Full ...   78000     4.8     0   
137  Drip Coffee 10g Arabica Bali Kintamani (4 Sachet)   65000     5.0     0   
151    Drip Coffee 10g Arabica Java Ciwidey (4 Sachet)   65000     NaN     0   
153  Drip Coffee 10g Arabica Mandheling Honey Proce...   65000     NaN     0   
154      Drip Coffee 10g Arabica Sidikalang (4 Sachet)   65000     5.0     0   
155  Refill Cold Brew - Arabica Aceh Gayo Atu Linta...   78000     NaN     0   
160         Drip Coffee 10g Robusta Lampung (4 Sachet)   60000     NaN     0   
161  Drip Coffee 10g Arabica Aceh Gayo Natural Proc...   65000     NaN     0   
162  Drip Coffee 10g Arabica Lintong Onan Ganjang (...   65000     NaN     0   
165                 Gewor Manglayang 200g Kopi Arabica  106200     5.0     0   
166                      Pana Mamasa 200g Kopi Arabica  106200     4.0     0   
167       Costa Rica Hacienda Sonora 150g Kopi Arabica  135000     4.0     0   
168                  Ethiopia Danche 150g Kopi Arabica  135000     5.0     0   
169                      Papua Sabin 200g Kopi Arabica  138600     NaN     0   
184  Drip Coffee 10g Arabica Bali Natural Process (...   65000     NaN     0   
185  Green Bean Kopi Arabica Java Ijen Red Honey Pr...  195000     NaN     0   
189  Refill Cold Brew - Arabica Flores Manggarai Fu...   78000     NaN     0   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN     0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN     0   

                                                 Image  \
54   https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
110  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
114  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
122  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
137  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
151  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
153  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
154  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
155  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
160  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
161  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
162  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
165  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
166  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
167  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
168  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
169  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
184  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
185  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
189  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
190  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
54   /coffee-beans/argopuro-anaerobic-natural-200g-...  
110  /coffee-beans/peaberry-500g-kopi-arabica?item_...  
114  /coffee-beans/java-ciwidey-honey-process-500g-...  
122  /cold-brew/cold-brew-arabica-flores-bajawa-ful...  
137  /drip-coffee/drip-coffee-10g-arabica-bali-kint...  
151  /drip-coffee/drip-coffee-10g-arabica-java-ciwi...  
153  /drip-coffee/drip-coffee-10g-arabica-mandhelin...  
154  /drip-coffee/drip-coffee-10g-arabica-sidikalan...  
155  /cold-brew/cold-brew-arabica-aceh-gayo-atu-lin...  
160  /drip-coffee/drip-coffee-10g-robusta-lampung-4...  
161  /drip-coffee/drip-coffee-10g-arabica-gayo-natu...  
162  /drip-coffee/drip-coffee-10g-arabica-lintong-o...  
165  /coffee-beans/gewor-manglayang-200g-kopi-arabi...  
166  /coffee-beans/pana-mamasa-200g-kopi-arabica?it...  
167  /coffee-bea

Berdasarkan data di atas, terdapat beberapa data pada kolom sold yang bernilai 0, akan tetapi pada kolom rating di baris yang sama, valuenya masih bisa diambil.
Hal tersebut terjadi karena pada website yang dilakukan scraping terdapat dua hal berbeda, ketika stok pada website masih banyak, maka akan muncul data berapa banyak yang sudah terjual. Akan tetapi ketika stok tersisa sedikit, akan dimunculkan data sisa stok yang tersedia pada toko.

In [73]:
df[df['Rating'].isnull()]

Product   Price  Rating  Sold  \
132  Green Bean Kopi Arabica Bali Ulian Natural - 1 Kg  190000     NaN   273   
138    Drip Coffee 10g Arabica Papua Wamena (4 Sachet)   75000     NaN   530   
143  Drip Coffee 10g Arabica Java Ciwidey Honey Pro...   70000     NaN   145   
146  Drip Coffee 10g Arabica Flores Manggarai Natur...   70000     NaN   109   
147  Drip Coffee 10g Arabica Java Sindoro Natural P...   70000     NaN    16   
151    Drip Coffee 10g Arabica Java Ciwidey (4 Sachet)   65000     NaN     0   
153  Drip Coffee 10g Arabica Mandheling Honey Proce...   65000     NaN     0   
155  Refill Cold Brew - Arabica Aceh Gayo Atu Linta...   78000     NaN     0   
157  Drip Coffee 10g Arabica Java Wanoja Kamojang N...   70000     NaN     7   
158       Green Bean Kopi Arabica Bali Ulian FW - 1 Kg  180000     NaN   124   
159  Green Bean Kopi Arabica Java Ciwidey Honey Pro...  200000     NaN    88   
160         Drip Coffee 10g Robusta Lampung (4 Sachet)   60000     NaN     0   
161  Drip Coffee 10g Arabica Aceh Gayo Natural Proc...   65000     NaN     0   
162  Drip Coffee 10g Arabica Lintong Onan Ganjang (...   65000     NaN     0   
164      Drip Coffee 10g Arabica Mandheling (4 Sachet)   65000     NaN   655   
169                      Papua Sabin 200g Kopi Arabica  138600     NaN     0   
184  Drip Coffee 10g Arabica Bali Natural Process (...   65000     NaN     0   
185  Green Bean Kopi Arabica Java Ijen Red Honey Pr...  195000     NaN     0   
189  Refill Cold Brew - Arabica Flores Manggarai Fu...   78000     NaN     0   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN     0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN     0   

                                                 Image  \
132  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
138  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
143  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
146  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
147  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
151  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
153  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
155  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
157  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
158  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
159  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
160  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
161  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
162  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
164  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
169  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
184  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
185  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
189  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
190  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
132  /green-bean/green-bean-kopi-arabica-bali-ulian...  
138  /drip-coffee/drip-coffee-10g-arabica-papua-wam...  
143  /drip-coffee/drip-coffee-10g-arabica-java-ciwi...  
146  /drip-coffee/drip-coffee-10g-arabica-flores-ma...  
147  /drip-coffee/drip-coffee-10g-arabica-java-sind...  
151  /drip-coffee/drip-coffee-10g-arabica-java-ciwi...  
153  /drip-coffee/drip-coffee-10g-arabica-mandhelin...  
155  /cold-brew/cold-brew-arabica-aceh-gayo-atu-lin...  
157  /drip-coffee/drip-coffee-10g-arabica-java-wano...  
158  /green-bean/green-bean-kopi-arabica-bali-ulian...  
159  /green-bean/green-bean-kopi-arabica-java-ciwid...  
160  /drip-coffee/drip-coffee-10g-robusta-lampung-4...  
161  /drip-coffee/drip-coffee-10g-arabica-gayo-natu...  
162  /drip-coffee/drip-coffee-10g-arabica-lintong-o...  
164  /drip-coffee/drip-coffee-10g-arabica-mandhelin...  
169  /coffee-beans/papua-sabin-200g-kopi-arabica?it...  
184  /drip-coffee/drip-coffee-10g-arabic

Berdasarkan data di atas, terdapat beberapa baris pada kolom Rating yang nilainya NaN, hal tersebut terjadi masih karena hal alami, karena memang banyak pembeli yang telah melakukan pembelian, masih belum atau tidak juga melakukan rating setelah melakukan pembelian. Serta barang yang nilai terjualnya masih 0, maka akan otomatis belum ada nilai rating. Mengacu pada dua hal tersebut, data tidak perlu dihapus karena penting untuk analisa data.

Selanjutnya dilakukan analisis statistik data pada DataFrame

In [76]:
# Menampilkan deskripsi statistik dari kolom numeric
df.describe()

Price      Rating           Sold
count     195.000000  174.000000     195.000000
mean   123806.153846    4.893103    8430.071795
std     59164.855418    0.128839   27200.221786
min     42750.000000    4.000000       0.000000
25%     74500.000000    4.900000     317.000000
50%    106200.000000    4.900000    1900.000000
75%    169200.000000    5.000000    6400.000000
max    400000.000000    5.000000  326100.000000

Berikut merupakan analisis dari data di atas:

Price

1. Memiliki value rata-rata harga kopi yang ada pada DataFrame yakni senilai Rp 123.806,-
2. Memiliki value standar deviasi yang ada pada DataFrame yakni senilai Rp 59.164,-
3. Memiliki value harga terendah dari kopi yang ada pada DataFrame yakni senilai Rp 42.750,-
4. Memiliki value harga tertinggi dari kopi yang ada pada DataFrame yakni senilai Rp 400.000,-


Rating

1. Memiliki value rata-rata rating kopi yang ada pada DataFrame senilai 4.89
2. Memiliki value standar deviasi yang ada pada DataFrame yakni senilai 0,12
3. Memiliki value rating terendah dari kopi yang ada pada DataFrame yakni senilai 4,0
4. Memiliki value rating tertinggi dari kopi yang ada pada DataFrame yakni senilai 5,0


Sold

1. Memiliki value rata-rata terjualnya kopi yang ada pada DataFrame sejumlah 8.430
2. Memiliki value standar deviasi yang ada pada DataFrame yakni sejumlah 27.200
3. Memiliki value terjualnya terendah dari kopi yang ada pada DataFrame yakni sejumlah 0
4. Memiliki value terjualnya tertinggi dari kopi yang ada pada DataFrame yakni sejumlah 326.100

Eksplorasi data yang memiliki value tertinggi dan terendah

In [77]:
df.groupby(['Product'])['Price'].sum().sort_values(ascending=False).head()

Product
Kopi Luwak Arabica 250g Pouch                         400000
Flores Manggarai Natural Process 500g Kopi Arabica    277200
Java Ijen Natural Process 500g Kopi Arabica           276000
Toraja Sapan 500g Kopi Arabica                        259200
Java Ijen Red Honey Process 500g Kopi Arabica         256000
Name: Price, dtype: int64

Berdasarkan data di atas, 5 produk dengan harga jual tertinggi yakni:
1. Kopi Luwak Arabica 250g Pouch dengan harga Rp 400.000,-
2. Flores Manggarai Natural Process 500g Kopi Arabica dengan harga Rp 277.200,-
3. Java Ijen Natural Process 500g Kopi Arabica dengan harga Rp 276.000,-
4. Toraja Sapan 500g Kopi Arabica dengan harga Rp 259.500,-
5. Java Ijen Red Honey Process 500g Kopi Arabica dengan harga Rp 256.000,-

In [80]:
df.groupby(['Product'])['Price'].sum().sort_values(ascending=True).head()

Product
Cappuccino Del Italiano - Kopi House Blend 200gr                    42750
Kopi Susu Blend - 200gr                                             46350
Biji Kopi Arabica Colombia Finca El Diviso Ethiopia Landrace 15g    48000
Biji Kopi Arabica Colombia Finca El Diviso Sidra Natural 15g        52000
Drip Coffee 10g Robusta Aceh Gayo (4 Sachet)                        55000
Name: Price, dtype: int64

Berdasarkan data di atas, 5 produk dengan harga jual terendah yakni:
1. Cappuccino Del Italiano - Kopi House Blend 200gr dengan harga Rp 42.750,-
2. Kopi Susu Blend - 200gr dengan harga Rp 46.350,-
3. Biji Kopi Arabica Colombia Finca El Diviso Ethiopia Landrace 15g dengan harga Rp 48.000,-
4. Biji Kopi Arabica Colombia Finca El Diviso Sidra Natural 15g dengan harga Rp 52.000,-
5. Drip Coffee 10g Robusta Aceh Gayo (4 Sachet) dengan harga Rp 55.000,-

In [81]:
df.groupby(['Product'])['Sold'].sum().sort_values(ascending=False).head()

Product
Crema Espresso - Kopi House Blend 500gr     326100
Kopi Susu Blend - 500gr                     137200
Kopi Gold Blend - 500gr                      90100
Aceh Gayo Atu Lintang 200g Kopi Arabica      57000
Premium Arabica - Kopi House Blend 500gr     55900
Name: Sold, dtype: int64

Berdasarkan data di atas, 5 produk dengan jumlah terjual tertinggi yakni:
1. Crema Espresso - Kopi House Blend 500gr dengan jumlah terjual sebanyak 326.100
2. Kopi Susu Blend - 500gr dengan jumlah terjual sebanyak 137.200
3. Kopi Gold Blend - 500gr Kopi Arabica dengan jumlah terjual sebanyak 90.100
4. Aceh Gayo Atu Lintang 200g Kopi Arabica dengan jumlah terjual sebanyak 57.000
5. Premium Arabica - Kopi House Blend 500gr dengan jumlah terjual sebanyak 55.900

In [82]:
df.groupby(['Product'])['Sold'].sum().sort_values(ascending=True).head()

Product
Drip Coffee 10g Arabica Bali Natural Process (4 Sachet)     0
Refill Cold Brew - Arabica Aceh Gayo Atu Lintang Natural    0
Peaberry 500g Kopi Arabica                                  0
Ethiopia Danche 150g Kopi Arabica                           0
Papua Sabin 200g Kopi Arabica                               0
Name: Sold, dtype: int64

Berdasarkan data di atas, 5 produk dengan jumlah terjual terendah yakni:
1. Drip Coffee 10g Arabica Bali Natural Process (4 Sachet) dengan jumlah sebanyak 0
2. Refill Cold Brew - Arabica Aceh Gayo Atu Lintang Natural dengan jumlah sebanyak 0
3. Peaberry 500g Kopi Arabica dengan jumlah sebanyak 0
4. Ethiopia Danche 150g Kopi Arabica dengan jumlah sebanyak 0
5. Papua Sabin 200g Kopi Arabica dengan jumlah sebanyak 0

In [83]:
len(df[df['Product'].str.contains('Arabica')])

143

Berdasarkan data di atas, dapat disimpulkan terdapat 143 kopi jenis Arabica yang dijual di Otten Coffee

In [84]:
len(df[df['Product'].str.contains('Robusta')])

24

Berdasarkan data di atas, dapat disimpulkan terdapat 24 kopi jenis Robusta yang dijual di Otten Coffee

In [85]:
len(df[df['Product'].str.contains('Drip')])

24

Berdasarkan data di atas, dapat disimpulkan terdapat 24 kopi jenis Drip Coffee yang dijual di Otten Coffee

In [86]:
len(df[df['Product'].str.contains('Ready to Drink')])

2

Berdasarkan data di atas, dapat disimpulkan terdapat 2 kopi jenis Ready to Drink yang dijual di Otten Coffee

In [87]:
len(df[df['Product'].str.contains('Capsule')])

21

Berdasarkan data di atas, dapat disimpulkan terdapat 21 kopi jenis Capsule yang dijual di Otten Coffee

In [88]:
df

Product   Price  Rating  \
0              Crema Espresso - Kopi House Blend 500gr  111600     4.9   
1    Kopi Susu Blend 1kg Biji Bubuk Kopi Arabica Ro...  151200     4.9   
2    Premium Espresso Blend 1kg Biji Bubuk Kopi Ara...  143100     4.9   
3                              Kopi Susu Blend - 500gr   83700     4.9   
4                              Kopi Gold Blend - 500gr   93600     4.9   
..                                                 ...     ...     ...   
190     Refill Cold Brew - Arabica Lintong Semi Washed   78000     NaN   
191      Refill Cold Brew - Arabica Mandheling Natural   78000     5.0   
192        Refill Cold Brew - Robusta Bali Full Washed   78000     NaN   
193  Cold Brew 1L - Arabica Aceh Gayo Semi Washed (...  198000     5.0   
194  Cold Brew 1L - Arabica Bali Kintamani Full Was...  198000     5.0   

       Sold                                              Image  \
0    326100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
1     18200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
2     15500  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
3    137200  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
4     90100  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
..      ...                                                ...   
190       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
191     172  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
192       0  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
193     107  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
194     105  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
0    /commercial-blend/crema-espresso-kopi-house-bl...  
1    /commercial-blend/kopi-susu-blend-1kg-arabica-...  
2    /commercial-blend/premium-espresso-blend-1kg-5...  
3    /commercial-blend/kopi-susu-blend-500gr?item_l...  
4    /commercial-blend/kopi-gold-blend-500gr?item_l...  
..                                                 ...  
190  /cold-brew/cold-brew-arabica-lintong-semi-wash...  
191  /cold-brew/cold-brew-arabica-mandheling-natura...  
192  /cold-brew/cold-brew-robusta-bali-refill?item_...  
193  /cold-brew/cold-brew-1l-arabica-aceh-gayo-semi...  
194  /cold-brew/cold-brew-1l-arabica-bali-kintamani...  

[195 rows x 6 columns]

In [89]:
df[df['Product'].str.contains('Arabica Toraja Sapan')]

Product   Price  Rating  Sold  \
28   Kopi Kapsul - O10 Arabica Toraja Sapan Coffee ...   74500     5.0  6400   
117    Drip Coffee 10g Arabica Toraja Sapan (4 Sachet)   70000     5.0   819   
124        Green Bean Kopi Arabica Toraja Sapan - 1 Kg  220000     4.8   679   
180  Refill Cold Brew - Arabica Toraja Sapan Semi W...   78000     5.0   383   

                                                 Image  \
28   https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
117  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
124  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   
180  https://d8g5mz6srwlcs.cloudfront.net/thumbnail...   

                                                  Link  
28   /capsules/kopi-kapsul-o10-arabica-toraja-sapan...  
117  /drip-coffee/drip-coffee-10g-arabica-toraja-sa...  
124  /green-bean/green-bean-kopi-arabica-toraja-sap...  
180  /cold-brew/cold-brew-arabica-toraja-sapan-semi...

Sekian analisis data telah berhasil dilakukan

# Data Storage

In [91]:
df.to_csv('otten_coffee.csv', index=False)    # index=False -> agar kolom index tidak ikut masuk ke dalam file .csv